lang/cs/Org.Apache.REEF.Wake/Remote/IType.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Org.Apache.REEF.Wake.Remote
+{
+    /// <summary>
+    /// Interface that classes should implement to express their type
+    /// </summary>
+    public interface IType
```

Comment:
Is IType and IWritable always be used together? In that case, you can put ClassType()method in IWritable and remove IType.